### ISSUE TITLE
Extended Desktop Size sending an FB update for every FB request with incremental=0 or just first

### ISSUE THREAD

**Issue by berrange** [rfbproto/rfbproto#37]
Reading the Extended Desktop Size spec

https://github.com/rfbproto/rfbproto/blob/master/rfbproto.rst#extendeddesktopsize-pseudo-encoding

This paragraph sticks out:

  "The server must send an ExtendedDesktopSize rectangle in response to a FramebufferUpdateRequest with incremental set to zero, assuming the client has requested the ExtendedDesktopSize pseudo-encoding using the SetEncodings message. This requirement is needed so that the client has a reliable way of fetching the initial screen configuration, and to determine if the server supports the SetDesktopSize message."

The justification for sending an ExtendedDesktopSize in response to a FramebufferUpdateRequest with incremental=0 makes sense as a mechanism for fetching initial configuration, and for feature detection,  for the first FramebufferUpdateRequest sent.

Once an app has detected existance of the feature, or fetched the initial configuration, why do subsequent requests with incremental=0 need to trigger an ExtendedDesktopSize rect too ?  If only the first FramebufferUpdateRequest triggered this behaviour, it would trivially avoid any risk of infinite loops in the clients.

I queried this behaviour in QEMU because the proposed patch for QEMU does *not* correctly follow this part of the spec in its proposed impl of Extended Desktop Size:

  https://lists.gnu.org/archive/html/qemu-devel/2020-12/msg03937.html

Instead of sending a ExtendedDesktopSize in response to FramebufferUpdateRequest with incremental=0,  QEMU unconditionally sends ExtendedDesktopSize rect after the SetEncodings call has indicated that the client supports ExtendedDesktopSize encoding.

Is there some scenario I'm missing that can only be solved by sending ExtendedDesktopSize in response to *every* request with incremental=0 ?

**Comment by CendioOssman** [comment]
It's been too long so unfortunately I don't remember the details. The goal at least was to reduce state. With the current way the server doesn't have to remember if this message has already been sent or not. Similarly, the client can re-request the layout in case it has dropped all knowledge about it. Similar to how it re-request the entire screen content because it has dropped it.

I'm afraid I don't remember a specific issue to point to, but I think there were similar issue historically that we wanted to avoid going forward.

Is it difficult to get QEMU compliant here? If QEMU currently sends a `FrameBufferUpdate` as a result of a `SetEncodings` then that is extremely out of spec, not just with regard to `ExtendedDesktopSize` and can have other weird effects. Unsolicited updates are not allowed.

**Comment by berrange** [comment]
I don't think it should be that difficult to always send an ExtendedDesktopSize rect in response to incremental=0 as that code path looks quite simple to me.

I have a feeling QEMU has broader pre-existing problems wrt unsolicited rects for psuedo-encodings, as it handles sending of many psuedo-encodings completely separately from sending of real framebuffer rects.
